test/protocol/insurance.js

```diff
@@ -0,0 +1,271 @@
+"use strict";
+
+const assertEx = require("../testutils.js");
+
+const Amber = artifacts.require("./Amber.sol");
+const Market = artifacts.require("./protocol/Market/Market.sol");
+const Offer = artifacts.require("./protocol/Market/Offer.sol");
+const MeasurementsOnChain = artifacts.require("./protocol/Measurements/MeasurementsOnChain.sol");
+const RangeRequirementsFactory = artifacts.require("./protocol/Requirements/RangeRequirementsFactory.sol");
+const RangeValidator = artifacts.require("./protocol/Validator/RangeValidator.sol");
+const EscrowedAgreement = artifacts.require("./protocol/Agreement/EscrowedAgreement.sol");
+const Insurance = artifacts.require("./protocol/Insurance/Insurance.sol");
+
+contract('Insurance', function(accounts) {
+    var amber;
+    var agreement;
+    var insurance;
+    
+    var market, offer, measurements, requirements, validator;
+    var buyer, seller, amberOwner, marketOwner, insuranceOwner;
+
+    var newAgreementInsuredListener;
+    var reimbursementRequestedListener;
+
+    beforeEach("deploy contracts", async() => {
+        buyer = accounts[1];
+        seller = accounts[2];
+        amberOwner = accounts[9];
+        marketOwner = accounts[8];
+        insuranceOwner = accounts[7];
+
+        let result = await web3.eth.getBlock('earliest');
+        let startTime = result.timestamp;
+        amber = await Amber.new(startTime, startTime, {from: amberOwner});
+
+        market = await Market.new(amber.address, marketOwner);
+        measurements = await MeasurementsOnChain.new({from: seller});
+        requirements = await RangeRequirementsFactory.new("name", market.address, {from: seller});
+        validator = await RangeValidator.new(measurements.address, requirements.address, {from: seller});
+
+        offer = await Offer.new("Fish", "Norway", "shark", "Qma", 40, 300, market.address, seller, measurements.address, requirements.address, validator.address, {from: seller});
+
+        var amount = await offer.priceFor(42);
+        await amber.mintLiquidToken(buyer, amount, {from: amberOwner});
+        await amber.approve(market.address, amount, {from: buyer});
+
+        await market.buy(offer.address, 42, {from: buyer});
+        agreement = await market.getNewestAgreement({from: buyer});
+
+        insurance = await Insurance.new(amber.address, {from: insuranceOwner});
+        newAgreementInsuredListener = await insurance.NewAgreementInsured();
+        reimbursementRequestedListener = await insurance.ReimbursementRequested();
+    });
+
+    describe('INSURING AGREEMENT', () => {
+        it("should not insure null agreement", async() => {
+            await assertEx.isReverted(async() => insurance.insure(0, 42, {from: seller}));
```

Comment:
I would consider using:
https://github.com/OpenZeppelin/zeppelin-solidity/blob/5751eac44401f44ca9d27d396341ea7ac1f73f46/test/helpers/assertRevert.js

Reply:
or
https://github.com/OpenZeppelin/zeppelin-solidity/blob/5751eac44401f44ca9d27d396341ea7ac1f73f46/test/helpers/expectThrow.js

Reply:
Actually neither of the existing functions gives me the functionality I need. As discussed I will be proposing enhancing the Zeppelin repository with the functions that I  need. If the proposal is accepted we can remove the code from this repo and use the ones from Zeppelin.